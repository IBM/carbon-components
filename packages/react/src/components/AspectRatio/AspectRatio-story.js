/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { storiesOf } from '@storybook/react';
import AspectRatio from '../AspectRatio';

function DemoContainer({ children }) {
  return <div style={{ width: 300, height: 300 }}>{children}</div>;
}

function DemoObject() {
  return (
    <div style={{ backgroundColor: 'black', width: '100%', height: '100%' }} />
  );
}

storiesOf('AspectRatio', module)
  .add('default', () => (
    <DemoContainer>
      <AspectRatio>
        <DemoObject />
      </AspectRatio>
    </DemoContainer>
  ))
  .add('16x9', () => (
    <DemoContainer>
      <AspectRatio ratio="16x9">
        <DemoObject />
      </AspectRatio>
    </DemoContainer>
  ))
  .add('2x1', () => (
    <DemoContainer>
      <AspectRatio ratio="2x1">
        <DemoObject />
      </AspectRatio>
    </DemoContainer>
  ))
  .add('4x3', () => (
    <DemoContainer>
      <AspectRatio ratio="4x3">
        <DemoObject />
      </AspectRatio>
    </DemoContainer>
  ));
