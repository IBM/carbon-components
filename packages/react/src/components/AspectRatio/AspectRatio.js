/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { settings } from 'carbon-components';
import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

const ratios = ['16x9', '2x1', '4x3', '1x1', '1x2'];
const { prefix } = settings;

function AspectRatio({
  className: customClassName,
  children,
  ratio = '1x1',
  ...rest
}) {
  const className = cx({
    [`${prefix}--aspect-ratio`]: true,
    [`${prefix}--aspect-ratio--${ratio}`]: true,
    [customClassName]: !!customClassName,
  });
  return (
    <div className={className} {...rest}>
      <div className="bx--aspect-ratio--object">{children}</div>
    </div>
  );
}

AspectRatio.propTypes = {
  /**
   * Specify an optional className to be applied to the container node
   */
  className: PropTypes.string,

  /**
   * Pass in the children that will be rendered within the aspect ratio object
   */
  children: PropTypes.node,

  /**
   * Specify the aspect ratio for the container
   */
  ratio: PropTypes.oneOf(ratios),
};

export default AspectRatio;
