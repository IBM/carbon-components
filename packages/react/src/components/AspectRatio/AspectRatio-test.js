/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { mount } from 'enzyme';
import React from 'react';
import AspectRatio from '../AspectRatio';

describe('AspectRatio', () => {
  it('should render', () => {
    const wrapper = mount(
      <AspectRatio>
        <div />
      </AspectRatio>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should support a custom `className` prop', () => {
    const wrapper = mount(
      <AspectRatio className="custom-class">
        <div />
      </AspectRatio>
    );
    expect(wrapper.children().find('.custom-class').length).toBe(1);
  });

  it.each(['16x9', '2x1', '4x3', '1x1', '1x2'])(
    'should support an aspect ratio of %s',
    ratio => {
      const wrapper = mount(
        <AspectRatio ratio={ratio}>
          <div />
        </AspectRatio>
      );
      // Rough heuristic to check for the bx--aspect-ratio--WxH class name on
      // the containing node
      expect(wrapper.children().prop('className')).toEqual(
        expect.stringContaining(ratio)
      );
    }
  );
});
