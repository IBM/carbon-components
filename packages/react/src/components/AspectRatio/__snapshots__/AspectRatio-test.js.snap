// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AspectRatio should render 1`] = `
<AspectRatio>
  <div
    className="bx--aspect-ratio bx--aspect-ratio--1x1"
  >
    <div
      className="bx--aspect-ratio--object"
    >
      <div />
    </div>
  </div>
</AspectRatio>
`;
